src/App.tsx

```diff
@@ -1,18 +1,33 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const optionArray = [3, 5, 10, 20];
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+  let startItem = 0;
+
+  if (currentPage > 1) {
+    startItem = perPage * (currentPage - 1);
+  }
+
+  let lastItem = startItem + perPage;
+
+  if (lastItem > 42) {
+    lastItem = 42;
+  }
```

Comment:
It's better to divide your code and move this logic to useEffect hook

Reply:
we don't learn this hook(

Reply:
https://mate.academy/chats/20588?selectedMessageId=851295
we just skip this comment in discusion in tread, because now it just not nececcary